src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs

```
@@ -153,7 +153,10 @@ public async Task<(Index Index, bool HasCreatedAutoIndex)> CreateAutoIndexIfNeed
             while (TryMatchExistingIndexToQuery(map, out index) == false)
             {
                 if (createAutoIndexIfNoMatchIsFound == false)
-                    throw new IndexDoesNotExistException("Could not find index for a given query.");
+                    throw new IndexDoesNotExistException("No Auto Index matching the given query was found.");
+
+                if (query.DisableAutoIndexCreation)
```

shouldn't we use `createAutoIndexIfNoMatchIsFound` here?

---

No, `createAutoIndexIfNoMatchIsFound ` has different purpose. We want to throw a different exception with better error message (`IndexDoesNotExistException` is translated to `404` status code later on - so it just shows `Index not found` in the studio)